### PR TITLE
Ports `slice` to TRT dialect

### PR DESCRIPTION
**Pull request by pranavm-nvidia**
Depends on https://github.com/NVIDIA/TensorRT-Incubator/pull/527

[Implements __abs__ for tensors](https://github.com/NVIDIA/TensorRT-Incubator/commit/58e499987e9d63df5000fb21d3169ef9253d99ae)

[Ports slice to TRT dialect](https://github.com/NVIDIA/TensorRT-Incubator/commit/aa7706611c338db37699986f0785978c882c8d06)

### CODE REVIEW

File: tripy/nvtripy/frontend/ops/slice.py [hunk 1]
```diff
@@ -63,151 +64,82 @@ def __getitem__(
...
+            steps.append(1)
+
+            squeeze_dims.append(dim_idx)
+        else:
+            assert isinstance(slice_idx, slice)
+
+            start = to_positive_idx(slice_idx.start) if slice_idx.start is not None else 0
+            step = utils.utils.default(slice_idx.step, 1)
+
+            # TODO (pranavm): Add exhaustive testing for negative/positive start/stop/step and 1/non-1 step and tensor/non-tensor params.
```
Comment (yizhuoz004) [inline tripy/nvtripy/frontend/ops/slice.py:110]: Is my understanding correct that we just pass `step` to tensorrt, even if it is negative?

**Review by yizhuoz004 — APPROVED** [review]

File: tripy/nvtripy/frontend/ops/slice.py (hunk 1 above)
Comment (pranavm-nvidia) [reply tripy/nvtripy/frontend/ops/slice.py:110]: Yes, TRT can handle negative `step`s. The only tricky part here is computing the `size` based on the `step`.
